### ISSUE TITLE
wxGLCanvasEGL::CreateSurface() broken by Nvidia 390.xxx driver

### ISSUE THREAD

**Issue by bahnwaerter** [wxWidgets/wxWidgets#23472]
### Description
If an EGL surface is created by the `wxGLCanvasEGL::CreateSurface()` method call, the surface creation fails on a platform using the legacy Nvidia 390.xxx driver. The failure can be tackled down to a broken `eglCreatePlatformWindowSurface(...)` function call. It seems like the legacy Nvidia 390.xxx driver for older graphics cards does not implement the `eglCreatePlatformWindowSurface` function properly. This has also been observed and reported by others (see https://forums.developer.nvidia.com/t/390-143-eglcreateplatformwindowsurface-causes-bogus-egl-bad-display-error/179820).

The specific bug was observed when trying to turn on graphics acceleration in KiCad 7.0.1. A detailed description can be found in the associated [KiCad issue](https://gitlab.com/kicad/code/kicad/-/issues/14570).

### Platform and version information

- wxWidgets version: 3.2.2.1
- wxWidgets port: wxGTK
- OS: Arch Linux (Kernel 6.1.24-1-lts, x86-64)

- GTK version: 3.24.37
- GDK backend: X11

- Graphics card: Nvidia GeForce GT630
- Graphics driver: Nvidia 390.157 (legacy driver for older GeForce cards)

**Comment by bahnwaerter** [comment]
Switching from the broken `eglCreatePlatformWindowSurface(...)` call to the alternative `eglCreateWindowSurface(...)` function call fixes the problem if wxWidgets is built with EGL support. To do this, wxWidgets 3.2.2.1 must be patched as follows:
```diff
diff --git a/src/unix/glegl.cpp b/src/unix/glegl.cpp
index ee6807b..8b37f23 100644
--- a/src/unix/glegl.cpp
+++ b/src/unix/glegl.cpp
@@ -475,8 +475,8 @@ bool wxGLCanvasEGL::CreateSurface()
     if (wxGTKImpl::IsX11(window))
     {
         m_xwindow = GDK_WINDOW_XID(window);
-        m_surface = eglCreatePlatformWindowSurface(m_display, *m_config,
-                                                   &m_xwindow, NULL);
+        m_surface = eglCreateWindowSurface(m_display, *m_config,
+                                           (EGLNativeWindowType) m_xwindow, NULL);
         m_readyToDraw = true;
     }
 #endif
@@ -507,8 +507,8 @@ bool wxGLCanvasEGL::CreateSurface()
         wl_surface_set_buffer_scale(m_wlSurface, scale);
         m_wlEGLWindow = wl_egl_window_create(m_wlSurface, w * scale,
                                              h * scale);
-        m_surface = eglCreatePlatformWindowSurface(m_display, *m_config,
-                                                   m_wlEGLWindow, NULL);
+        m_surface = eglCreateWindowSurface(m_display, *m_config,
+                                           (EGLNativeWindowType) m_wlEGLWindow, NULL);
         m_wlFrameCallbackHandler = wl_surface_frame(surface);
         wl_callback_add_listener(m_wlFrameCallbackHandler,
                                  &wl_frame_listener, this);
```
Note that this issue is not a wxWidgets (and KiCad) related bug, but seems to be an Nvidia-specific problem. That's why this provided patch is only a workaround. Further investigations would be very helpful to narrow down the problem. Currently I cannot make any assumption whether all older Nvidia graphics cards with this related legacy driver are affected.

**Comment by vadz** [comment]
Thanks for reporting this!

I don't know anything about EGL, but the [documentation](https://registry.khronos.org/EGL/sdk/docs/man/html/eglCreateWindowSurface.xhtml) seems to say that the 2 functions are identical except that there are fewer restrictions on type of the display/config with `eglCreateWindowSurface()`. If this is correct, it looks like we don't really lose anything by applying this patch, so we probably should do it.

@swt2c Any thoughts?

**Comment by bahnwaerter** [comment]
> I don't know anything about EGL, but the [documentation](https://registry.khronos.org/EGL/sdk/docs/man/html/eglCreateWindowSurface.xhtml) seems to say that the 2 functions are identical except that there are fewer restrictions on type of the display/config with eglCreateWindowSurface().

Correct. That was also the information that inspired me to switch to the `eglCreateWindowSurface(...)` function call. But let's be honest, I'm not an EGL specialist either.

Note that I've only tested this patch with X11 so far. A test with Wayland is still pending. In addition, the patch does not yet meet the code quality requirements (see code formatting). Once this is all done, I'm happy to create a clean pull request.

**Comment by swt2c** [comment]
I think it's probably OK to switch to `eglCreateWindowSurface()`.  I can't remember why I selected `eglCreatePlatformWindowSurface()` initially - probably just to the avoid the casts.  But they don't seem terrible.
